ceph_salt/config_shell.py

```diff
@@ -655,6 +661,124 @@ def ui_command_remove(self, value):
             PP.pl_red('Value not found.')
 
 
+class ConfElementNode(configshell.ConfigNode):
+    def __init__(self, key, value, parent):
+        configshell.ConfigNode.__init__(self, key, parent)
+        self.value = value
+
+    def summary(self):
+        return self.value, None
+
+
+class ConfSectionNode(OptionNode):
+    def __init__(self, option_name, option_dict, parent):
+        super(ConfSectionNode, self).__init__(option_name, option_dict, parent)
+        value_dict, _ = self._find_value()
+        self.value = dict(value_dict)
+        for setting, value in self.value.items():
+            ConfElementNode(setting, value, self)
+
+    def _list_commands(self):
+        return ['set', 'remove']
+
+    def summary(self):
+        return '', None
+
+    @staticmethod
+    def _normalize(text):
+        text = text.strip()
+        if text.startswith('"') and text.endswith('"'):
+            text = text[1:-1]
+        return text.strip()
+
+    def ui_command_set(self, expr):
+        """
+        Expression has "<setting> = <value>" format.
+
+        Example: set osd crush chooseleaf type = 0
+        """
+        separator_count = expr.count('=')
+        if separator_count != 1:
+            PP.pl_red("Invalid format, try 'set <setting> = <value>'.")
+            return
+        expr = self._normalize(expr)
+        setting, value = [self._normalize(s) for s in expr.split('=')]
+        child = None
+        if setting in self.value:
+            child = self.get_child(setting)
+        self.value[setting] = value
+        self.option_dict['handler'].save(self.value)
+        if child:
+            child.value = value
+        else:
+            ConfElementNode(setting, value, self)
+        PP.pl_green('Setting set.')
```

Comment:
"Setting set" sounds odd in English. Could it be rephrased?

I would prefer: "Setting added." This goes nicely with "Setting removed."

Reply:
Now that I think about it, "osd crush chooseleaf type = 0" is not a setting that can have a value. Here we are not "setting a setting", but rather adding a line to the bootstrap ceph.conf.

Would it make sense to change the syntax from "set" to "add", i.e.:

```
ceph-salt config /Deployment/Bootstrap_Ceph_Conf/global add osd crush chooseleaf type = 0
```

Then it becomes even more natural to write "Setting added."

Reply:
Feel free to ignore this comment. "Setting set" does sound a bit weird, but I'm prepared to live with it.

Maybe "Parameter set." / "Parameter removed." / "Parameter changed." would be nicer, instead of "Setting".